pkg/util/promtest/histogram.go

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package promtest
+
+import (
+	"fmt"
+	"slices"
+
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+)
+
+// IsNativeHistogram checks if a gathered metric has positive or negative spans.
+//
+// This is possible because native histograms without any observations and zero_threshold of 0
+// use a no-op span (offset 0, length 0).
+func IsNativeHistogram(g prometheus.Gatherer, metricName string) error {
+	family, err := gatherMetricFamily(g, metricName)
+	if err != nil {
+		return err
+	}
+	histograms, err := getHistograms(family)
+	if err != nil {
+		return err
+	}
+	for i, h := range histograms {
+		if len(h.PositiveSpan)+len(h.NegativeSpan) == 0 {
+			return fmt.Errorf("metric %d has no negative or positive spans", i)
+		}
+	}
+	return nil
+}
+
+// HasSampleCount checks if a gathered metric has a specific sample count.
+func HasSampleCount(g prometheus.Gatherer, metricName string, count float64) error {
+	family, err := gatherMetricFamily(g, metricName)
+	if err != nil {
+		return err
+	}
+	histograms, err := getHistograms(family)
+	if err != nil {
+		return err
+	}
+	for i, h := range histograms {
+		if h.SampleCountFloat != nil && *h.SampleCountFloat != count {
+			return fmt.Errorf("histogram %d should have %f samples but got %f", i, count, *h.SampleCountFloat)
+		}
+		if h.SampleCount != nil && *h.SampleCount != uint64(count) {
+			return fmt.Errorf("histogram %d should have %f samples but got %d", i, count, *h.SampleCount)
+		}
+	}
```

Comment:
It shouldn't happen but technically this code returns no error if there was no count at all, so let's do something super explicit like
```
switch {
  case h.SampleCountFloat != nil:
     if *h.SampleCountFloat != count {
       // error
     }
  case h.SampleCount != nil:
    if *h.SampleCount != uint64(count) {
      // error
    } 
  default:
     // error
}
```